common/k8stest/util_daemonset.go

```diff
@@ -109,3 +111,47 @@ func UpdateDemonsetContainerAllEnv(daemonsetName string, containerName string, n
 	}
 	return nil
 }
+
+func WaitForDaemonsetReady(dsName string, namespace string, sleepTime int, duration int) bool {
+	ready := false
+	count := (duration + sleepTime - 1) / sleepTime
+
+	logf.Log.Info("DaemonsetReadyCheck", "Daemonet", dsName, "namespace", namespace)
+	for ix := 0; ix < count && !ready; ix++ {
+		time.Sleep(time.Duration(sleepTime) * time.Second)
+
+		ready = DaemonSetReady(dsName, namespace)
+		logf.Log.Info("DaemonSetReady: ", "daemonset", dsName, "ready", ready)
+	}
+
+	if !ready {
+		logf.Log.Info("Daemonset not ready", "Daemonset", dsName, "namespace", namespace)
+		return false
+	}
+
+	// verify pod running
+	err := WaitForPodRunning(dsName, namespace, duration)
```

Comment:
Not convinced that this waits for all daemonset pods ( there can be multiple ) - that is the intention of this call right?

Reply:
yes, it's not waiting for all pods.
It's fixed now https://github.com/openebs/openebs-e2e/compare/3be0131d2bbe557531716cc1ea19de8e5669ef9b..0416e6a0469f9d2b971344e3025e3d2f16cd5e58